CHANGELOG.md

```diff
@@ -3,6 +3,205 @@
+  [Jan 2025](https://github.com/astral-sh/uv/pull/10634).
+
+  You can opt-out of this behavior with `uv python install --no-registry` or `UV_PYTHON_INSTALL_REGISTRY=0`.
+
+- **Prompt before removing an existing directory in `uv venv` ([#14309](https://github.com/astral-sh/uv/pull/14309))**
+
+  Previously, `uv venv` would remove an existing virtual environment without confirmation. While
+  this is consistent with the behavior of project commands, like `uv sync`, it's surprising to users
+  that are using imperative workflows, like `uv pip`. Now, `uv venv` will prompt for confirmation
+  before removing an existing virtual environment. **If not in an interactive context, uv will still
+  remove the virtual environment for backwards compatibility. However, this behavior is likely to
+  change in a future release.**
+
+  The behavior for other commands, like `uv sync`, is unchanged.
+
+  You can opt-out of this behavior by setting `UV_VENV_CLEAR=1` or passing the `--clear` flag.
+
+- **Validate that discovered interpreters meet the Python preference ([#7934](https://github.com/astral-sh/uv/pull/7934))**
+
+  uv allows opting out of its managed Python versions with the `--no-managed-python` and
+  `python-preference` options. Previously, uv would not enforce this option for Python interpreters
+  discovered on the `PATH`. For example, if a symlink to a managed Python interpreter was added, uv
+  would allow it to be used even if `--no-managed-python` was provided. Now, uv ignores Python
+  interpreters that do not conform to the Python preference _unless_ they are in an active virtual
+  environment or are explicitly requested, e.g., with `--python /path/to/python3.13`.
+
+  Similarly, uv would previously not invalidate existing project environments if they did not match
+  the Python preference. Now, uv will invalidate and recreate project environments when the Python
+  preference changes.
+
+  You can opt-out of this behavior by providing the explicit path to the Python interpreter.
+
+- **Install dependencies without build systems when they are `path` sources ([#14413](https://github.com/astral-sh/uv/pull/14413))**
+
+  When working a project, uv uses the [presence of a build
+  system](https://docs.astral.sh/uv/concepts/projects/config/#build-systems) to determine if it
+  should be built and installed into the environment. However, when a project is a dependency of
+  another project, it can be surprising for the dependency to be missing from the environment. Now,
+  dependencies with [`path` sources](https://docs.astral.sh/uv/concepts/projects/dependencies/#path)
+  are built and installed, regardless of the presence of a build system. If a build system is not
+  present, the `setuptools.build_meta:__legacy__ ` backend will be used (per the standards).
+
+  You can opt-out of this behavior by setting `package = false` in the source declaration.
```

Comment:
Is there a reason that one would want to opt out of this? Like I understand that this is breaking in the sense that `import foo` might succeed when it previously failed, but is there some common(-ish) case where people might run into this, like tests, or placeholders for foreign-language code, or something, that would be worth alluding to here?

Reply:
Your package might not build.